aws/prefect/prefect-agent-on-ecs/s3.tf

```diff
@@ -19,18 +19,30 @@ resource "aws_s3_bucket_public_access_block" "bucket_block" {
 }
 
 
-resource "aws_s3_object" "network_config" {
+resource "aws_s3_object" "task_definition" {
   depends_on = [
-    aws_s3_bucket.network_config
+    aws_s3_bucket.prefect_ecs_config
   ]
-  bucket = aws_s3_bucket.network_config.id
-  key    = "network-config.yaml"
-  content = templatefile("${path.module}/network-config.yaml.tftpl",
+  bucket = aws_s3_bucket.prefect_ecs_config.id
+  key    = "task-definition.yaml"
+  content = templatefile("${path.module}/task-definition.yaml.tftpl",
     {
-      subnet_ids         = var.subnet_ids,
-      security_group_ids = [aws_security_group.sg.id],
-      assign_public_ip   = "DISABLED",
+      region              = var.region
+      log_group_name      = var.flow_log_group_name
+      log_stream_prefix   = var.flow_log_stream_prefix
+      default_task_cpu    = var.default_task_cpu
+      default_task_memory = var.default_task_memory
     }
   )
-  etag = filemd5("${path.module}/network-config.yaml.tftpl")
-}
\ No newline at end of file
+  etag = filemd5("${path.module}/task-definition.yaml.tftpl")
+}
+
+resource "aws_s3_object" "network_config" {
+  depends_on = [
+    aws_s3_bucket.prefect_ecs_config
+  ]
+  bucket  = aws_s3_bucket.prefect_ecs_config.id
+  key     = "network-config.yaml"
+  content = yamlencode({ "networkConfiguration" : { "awsvpcConfiguration" : { "subnets" : [for subnet in var.subnet_ids : subnet], "securityGroups" : [aws_security_group.sg.id], "assignPublicIp" : "ENABLED" } } })
```

Comment:
just want to verify you tested this?  this is very particular and just want to make sure its been tested

Reply:
Yep, I am going to parameterize the public IP thing though. The client network I was working on was not in line with aws well architected design.